### PR TITLE
Alert: Sync to latest designs

### PR DESCRIPTION
**Pull request by philschanely**
## Description

This PR updates Alert to match latest design specs.

- [x] Heading icon aligns with heading text
- [x] Button and link colors use default colors and are not customizable
- [x] Ensure second link option is available in the button group

In addition, I documented some quesion and recommendations for team discussion to further improve this component: https://kajabi.atlassian.net/browse/DSS-169

I also have a follow up PR for cleaning up Rails documentation and welcome early feedback: https://github.com/Kajabi/sage-lib/pull/1600

## Screenshots

[Figma specs](https://www.figma.com/file/sMbtLUHSt2vfKgKjyQ3052/%F0%9F%A7%A9-Sage-components?node-id=13478%3A24453)

| | Before | After |
|---|---|---|
| Rails regular | Screen Shot 2022-09-30 at 12 05 44 PM | Screen Shot 2022-09-30 at 12 06 06 PM |
| Rails small | Screen Shot 2022-09-30 at 12 05 53 PM | Screen Shot 2022-09-30 at 12 06 13 PM |
| React regular | Screen Shot 2022-09-30 at 12 05 04 PM | Screen Shot 2022-09-30 at 12 07 19 PM |
| React small | Screen Shot 2022-09-30 at 12 04 56 PM | Screen Shot 2022-09-30 at 12 06 55 PM |

## Testing in `sage-lib`

- See http://localhost:4000/pages/component/alert?tab=preview
- See http://localhost:4100/?path=/docs/sage-alert--dismissable-alert

## Testing in `kajabi-products`

1. (**LOW**) Update Alert to latest design specs.

## Related

https://kajabi.atlassian.net/browse/DSS-143

### CODE REVIEW

**Comment by pixelflips** [comment]
@philschanely and I paired on this one for a quick and early review/discussion. Afterward, we agreed the icon updates should be separated out into their own PR to help with simplifying the PRs.

CC: @monicawheeler

File: docs/lib/sage_rails/app/sage_components/sage_alert.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class SageAlert < SageComponent
   set_attribute_schema({
-    color: [:optional, NilClass, SageSchemas::STATUSES],
+    color: [:optional, NilClass, Set.new(["default", "danger", "draft", "info", "locked", "success", "published", "warning", "reached", "exceeded", "approaching"])],
```
Comment (philschanely) [inline docs/lib/sage_rails/app/sage_components/sage_alert.rb:3]: FFR I think we should revisit these "color" values to better align with the design specs. There are more here than perhaps necessary.

File: docs/lib/sage_rails/app/views/sage_components/_sage_alert.html.erb [hunk 2]
```diff
@@ -42,7 +41,8 @@
             label: secondary_action[:value],
             show_label: true,
             style: "secondary",
-            url: secondary_action[:url]
+            remove_underline: true,
+            url: secondary_action[:url],
```
Comment (philschanely) [inline docs/lib/sage_rails/app/views/sage_components/_sage_alert.html.erb:45]: FFR we may benefit from adding `html_attibutes` here and match it in the Alert schema so engineers can provide other attributes for these links.

File: packages/sage-react/lib/Alert/Alert.jsx [hunk 3]
```diff
@@ -35,43 +36,29 @@ export const Alert = ({
     if (onDismiss) onDismiss();
   };
 
-  const renderIcon = () => {
-    let icon;
+  const getDefaultIcon = () => {
```
Comment (philschanely) [inline packages/sage-react/lib/Alert/Alert.jsx:39]: FFR: Should we add a similar aspect in the Rails component to ensure a default icon appears?

File: packages/sage-react/lib/Alert/Alert.jsx [hunk 4]
```diff
@@ -124,7 +112,7 @@ Alert.propTypes = {
   actions: PropTypes.node,
   className: PropTypes.string,
   color: PropTypes.oneOf(Object.values(ALERT_COLORS)).isRequired,
-  description: PropTypes.string,
+  description: PropTypes.node,
```
Comment (philschanely) [inline packages/sage-react/lib/Alert/Alert.jsx:115]: FFR: This expands what is allowed here so engineers can safely pass HTML content to the description in addition to string content.

**Review by pixelflips — APPROVED** [review]
Great work! LGTM! 👍🏼 

A couple of comments, nothing blocking. I did notice the `figma_embed` for Alerts may be loading an outdated URL for the Figma tab. Not a showstopper but might be an easy/quick fix.

File: docs/app/views/examples/components/alert/_preview.html.erb [hunk 5]
```diff
@@ -5,7 +5,8 @@
     launch: false,
     help_link: false,
     show_label: true,
-    style: "secondary"
+    style: "secondary",
+    remove_underline: true,
```
[the remark this answers is not in the document]
Comment (philschanely) [reply docs/app/views/examples/components/alert/_preview.html.erb:9]: I agree, @pixelflips. See point 5 on [this discussion task](https://kajabi.atlassian.net/browse/DSS-169) related to this in that if we indeed prefer `link_to` we may want to update the inner components used and look at whether we keep the open slot `alert_actions`.

**Review by monkeypox8 — APPROVED** [review]
👍 LGTM

Left a comment in the associated Jira issue on our approach for consolidating/deprecating props. As noted, there are a few areas to patch in KP once the required `icon` area goes into effect: mobile layout does not collapse as expected.

**Review by ju-Skinner — APPROVED** [review]

**Pull request merged**